source/overloading.tex

```diff
@@ -3260,29 +3260,32 @@
 \pnum
 A \defnadj{subscripting}{operator function}
 is a function named \tcode{\keyword{operator}[]}
-that is a non-static member function with exactly one parameter.
+that is a non-static member function.
```

Comment:
Should this say "with an arbitrary number of parameters", to be consistent with the previous subclause ([\[over.call\]](http://eel.is/c++draft/over.call#1.sentence-1))?

Reply:
@Mick235711, where do you think we disallow default arguments in the wording as presented?

Reply:
In 12.4.1 [over.oper.general], the 10-th clause is stated as follows:
> An operator function cannot have default arguments (9.3.4.7), except where explicitly stated below. Operator functions cannot have more or fewer parameters than the number required for the corresponding operator, as described in the rest of 12.4.

I think this clause is why that last clause (12.4.4, [over.call]) need to specify
> A function call operator function is a function named operator() that is a non-static member function with an arbitrary number of parameters. *It may have default arguments.* For an expression of the form

Therefore for consistency between `operator()` and `operator[]`, and be more clear with specification, I think we should specify `It may have default arguments` in [over.sub] too.

Reply:
Ok, thanks. This needs to become a core issue. Please file one.

Reply:
If anyone is interested in this: I cannot actually submit Core issues since the chair is not responding to anyone outside WG21 basically. Also, the compiler (GCC 12) really implements this as not accepting default arguments. So if anyone has the opportunity feel free to submit this as issue yourself.

Reply:
Sorry about that -- I think the CWG chair (Mike) may be slow to respond, but the issue shouldn't be lost, and will surface eventually. Thanks!